src/Orleans.Serialization/Configuration/DefaultTypeManifestProvider.cs

```diff
@@ -37,18 +37,24 @@ public void Configure(TypeManifestOptions typeManifest)
             wellKnownTypes[25] = typeof(int[]);
             wellKnownTypes[26] = typeof(string[]);
             wellKnownTypes[27] = typeof(Type);
+#if NET6_0_OR_GREATER
```

Comment:
These conditions are not going to work as expected on .NET 5 or 6.

Reply:
Perhaps we should add additional targets. There are likely some conditions which can be lowered to previous versions, like this one.

Reply:
I contemplated just using the `NET7_0_OR_GREATER` everywhere, but felt having conditions that better match the actual dependency made more sense. The net effect (since only 2 build targets exist) is as if only the single condition was used, but I can see how someone looking at the source might be confused.